File: android/core/src/main/kotlin/org/hdhropen/kit/models/DeviceInfo.kt
package org.hdhropen.kit.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class DeviceInfo(
    val id: String,
    val name: String
)

@Serializable
data class DeviceListEntry(
    val id: String,
    val name: String,
    @SerialName("last_seen_at")
    val lastSeenAt: String
)

@Serializable
data class DeviceRegisterResult(
    val id: String,
    val name: String,
    @SerialName("is_new")
    val isNew: Boolean = false
)

File: apple/HDHROpenKit/Sources/HDHROpenKit/Models/DeviceInfo.swift
import Foundation

public struct DeviceInfo: Identifiable, Codable, Sendable {
    public let id: String
    public let name: String
}

public struct DeviceListEntry: Identifiable, Codable, Sendable {
    public let id: String
    public let name: String
    public let lastSeenAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case lastSeenAt = "last_seen_at"
    }
}

public struct DeviceRegisterResult: Identifiable, Codable, Sendable {
    public let id: String
    public let name: String
    public let isNew: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case isNew = "is_new"
    }
}
